test_common/harness/testHarness.c

```diff
@@ -655,23 +658,29 @@ int callTestFunctions( test_definition testList[], unsigned char selectedTestLis
                        cl_command_queue_properties queueProps )
 {
     int totalErrors = 0;
+    const Version device_version = get_device_cl_version(deviceToUse);
 
     for( int i = 0; i < testNum; ++i )
     {
         if( selectedTestList[i] )
         {
-            // Skip unimplemented test (can happen when you select all of the tests)
-            if( testList[i].func != NULL )
+            if (testList[i].min_version > device_version)
             {
-                int errors = callSingleTestFunction( testList[i], deviceToUse, forceNoContextCreation,
-                                                     numElementsToUse, queueProps );
-                resultTestList[i] = errors;
-                totalErrors += errors;
+                log_info("Skipping %s. Device doesn't support functionality exercised by this test.\n",
```

Comment:
I would suggest making the message more specific. Probably something along the lines of "TEST requires at least version X but the device reported version Y".

Reply:
Need to rework how we deal with versions then.